### ISSUE TITLE
html? bootstrap ? form 크기 조정하는 방법 

### ISSUE THREAD

**Issue by cokemania2** [codingeverybody/codingyahac#734]
summernote` 라는 텍스트 에디터를 ejs파일에 넣어서 쓰려고하는데 
에디터가 width (너비가 ) 화면에 꽉 차게 나옵니다..
이거를 어떻게 줄여서 가운데에 놓을지 모르겠습니다..
검색을 어떻게해야할지도.. 
알려주시면 감사하겠습니다.

**Comment by cokemania2** [comment]
`<body>
<form action="/write_action" method="POST">
    
  <div class="form-group">
            <label for="title">제목:</label>
            <input type="text" class="form-control" id="title" name="title">
  </div>
  <div id="summernote"  class="click2edit" ><p> 이렇게 하면 됩니다 </p></div>

  <button type="submit" class="btn btn-success">등록하기</button>
</form>
  <script>
    $(document).ready(function() {
        $('#summernote').summernote({
            tabsize: 2,
            height: 300,
            lang: 'ko-KR' // 언어 세팅
        });
    });
  </script>
  </div>
</body>`

**Comment by Hatser** [comment]
```html
<div class="editor-container">
  <div id="summernote"  class="click2edit" ><p> 이렇게 하면 됩니다 </p></div>
</div>
```

```css
.editor-container {
  width: 70%;
  margin: 0 auto;
}
```

라이브러리를 이용했을 때 디자인 문제가 발생하면 우선 라이브러리가 요소들을 어떻게 처리하는지,
class이름은 무엇인지 확인할 필요가 있습니다.

확인해보니 이 라이브러리는
summernote를 실행하는 엘리먼트를 보이지않게 처리하고 그 형제 엘리먼트를 만들어 이 요소를 에디터로 이용하네요

```html
    <div id="summernote" class="click2edit" style="display: none;">...</div>
    <div class="note-editor note-frame panel panel-default">...</div>
```

위를 통해 생성되는 에디터 클래스네임을 알았으니

```css
.note-editor {

}
```

안에 속성을 부여해 디자인을 조절할 수도 있지만, 라이브러리 자체 css가 있는 만큼 충돌할 여지가 남아있습니다.

지금 상황에서는 크기와 위치만 조절하면 되니 간단하게 이를 감싸는 wrapper 엘리먼트를 만들어 이 wrapper 엘리먼트의 크기와 위치만 조절하면 해결할 수 있습니다.

**Comment by cokemania2** [comment]
답변정말 감사드립니다. 
그런데 제가 이런방식을 써본적이없어서 그런지 해도 적용이 안됩니다.
이렇게 쓰는게 맞나요 ?
`<form action="/write_action" method="POST">
    
  <div class="form-group">
            <label for="title">제목:</label>
            <input type="text" class="form-control" id="title" name="title">
  </div>
  <div id="summernote"  class="click2edit" ><p> 이렇게 하면 됩니다 </p></div>

  <button type="submit" class="btn btn-success">등록하기</button>
</form>.write-form {
    width: 70%;
    margin: 0 auto;
  }`

**Comment by Hatser** [comment]
```css
.write-form {
    width: 70%;
    margin: 0 auto;
}
```

부분은 CSS코드이므로 따로 .css 파일에 넣어서, HTML에서 불러오는 방식으로 사용하셔야 합니다.

```html
<form style="width: 70%;margin: 0 auto;" action="/write_action" method="POST">
    ...
</form>
```
처럼 사용하실 수도 있지만,

css를 사용하는 방식을 함께 익히시는 게 좋을 것 같아요

styles.css
```css
.write-form {
    width: 70%;
    margin: 0 auto;
}
```

~~~.html
```html
<head>
    <link href="styles.css" rel="stylesheet">
</head>
```

처럼 html파일에서 css파일을 불러와 사용할 수 있어요

```html
<head>
    <style>
        .write-form {
            width: 70%;
            margin: 0 auto;
        }
    </style>
</head>
```

처럼 html 파일 내부에 포함시키는 방법도 있습니다.

**Comment by cokemania2** [comment]
밖에 style.css 만들어서 해봤는데 
![image](https://user-images.githubusercontent.com/38974863/62482896-6326cf80-b7f1-11e9-9897-f3d643a97c94.png)

이렇게 전체화면을 계속 덮네요..
흙.. 제가 이해를 잘못한건가요ㅠㅠ 

전체코드입니다 

```
<!DOCTYPE html>
<html lang="en">

<head>
    <title>글쓰기</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    <!-- include libraries(jQuery, bootstrap) -->
    <link href="http://netdna.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.css" rel="stylesheet">
    <link href="styles.css" rel="stylesheet">
    <script src="http://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.js"></script> 
    <script src="http://netdna.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.js"></script> 

    <!-- include summernote css/js -->
    <link href="http://cdnjs.cloudflare.com/ajax/libs/summernote/0.8.12/summernote.css" rel="stylesheet">
    <script src="http://cdnjs.cloudflare.com/ajax/libs/summernote/0.8.12/summernote.js"></script>
</head>

<body>
<form action="/write_action" method="POST">
    
  <div class="form-group">
            <label for="title">제목:</label>
            <input type="text" class="form-control" id="title" name="title">
  </div>
  <div id="summernote"  class="click2edit" ><p> 이렇게 하면 됩니다 </p></div>

  <button type="submit" class="btn btn-success">등록하기</button>
</form>

</style>

  <script>
    $(document).ready(function() {
        $('#summernote').summernote({
            tabsize: 2,
            height: 300,
            lang: 'ko-KR' // 언어 세팅
        });
    });
  </script>
  </div>
</body>

</html>
```

**Comment by Hatser** [comment]
HTML
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <title>글쓰기</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    <!-- include libraries(jQuery, bootstrap) -->
    <link href="http://netdna.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.css" rel="stylesheet">
    <link href="styles.css" rel="stylesheet">
    <script src="http://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.js"></script> 
    <script src="http://netdna.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.js"></script> 

    <!-- include summernote css/js -->
    <link href="http://cdnjs.cloudflare.com/ajax/libs/summernote/0.8.12/summernote.css" rel="stylesheet">
    <script src="http://cdnjs.cloudflare.com/ajax/libs/summernote/0.8.12/summernote.js"></script>
</head>

<body>
<form class="write-form" action="/write_action" method="POST">
    
  <div class="form-group">
            <label for="title">제목:</label>
            <input type="text" class="form-control" id="title" name="title">
  </div>
  <div id="summernote"  class="click2edit" ><p> 이렇게 하면 됩니다 </p></div>

  <button type="submit" class="btn btn-success">등록하기</button>
</form>

  <script>
    $(document).ready(function() {
        $('#summernote').summernote({
            tabsize: 2,
            height: 300,
            lang: 'ko-KR' // 언어 세팅
        });
    });
  </script>
  </div>
</body>

</html>
```

styles.css
```css
.write-form {
    width: 70%;
    margin: 0 auto;
}
```

**Comment by cokemania2** [comment]
정말 감사합니다 !
